server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+@Api(tags = "Deleted Messages Vault")
+@Path(DeletedMessagesVaultRoutes.ROOT_PATH)
+@Produces(Constants.JSON_CONTENT_TYPE)
+public class DeletedMessagesVaultRoutes implements Routes {
+
+    static final String ROOT_PATH = "deletedMessages/user";
+    private static final String USER_PATH_PARAM = "user";
+    private static final String RESTORE_PATH = ROOT_PATH + SEPARATOR + ":" + USER_PATH_PARAM;
+
+    private final DeletedMessageVault deletedMessagesVault;
+    private final MailboxManager mailboxManager;
+    private final JsonTransformer jsonTransformer;
+    private final TaskManager taskManager;
+
+    @VisibleForTesting
+    DeletedMessagesVaultRoutes(DeletedMessageVault deletedMessagesVault, MailboxManager mailboxManager, JsonTransformer jsonTransformer,
+                               TaskManager taskManager) {
+        this.deletedMessagesVault = deletedMessagesVault;
+        this.mailboxManager = mailboxManager;
+        this.jsonTransformer = jsonTransformer;
+        this.taskManager = taskManager;
+    }
+
+    @Override
+    public String getBasePath() {
+        return ROOT_PATH;
+    }
+
+    @Override
+    public void define(Service service) {
+        service.post(RESTORE_PATH, this::restore, jsonTransformer);
+    }
+
+    @POST
+    @Path(ROOT_PATH)
+    @ApiOperation(value = "Restore deleted emails from a specified user to his new restore mailbox")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = true,
+            name = "user",
+            paramType = "path parameter",
+            dataType = "String",
+            defaultValue = "none",
+            example = "user@james.org",
+            value = "Compulsory. Needs to be a valid username represent for an user had requested to restore deleted emails")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.CREATED_201, message = "Task is created", response = TaskIdDto.class),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Bad request - user param is invalid"),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
+    })
+    private TaskIdDto restore(Request request, Response response) {
+        User userToRestore = extractUser(request);
+        TaskId taskId = taskManager.submit(new DeletedMessageVaultRestoreTask(deletedMessagesVault, userToRestore, mailboxManager));
```

Comment:
This class shouldn't have to worry about MailboxManager or deletedMessagesVault. Could you create a DeletedMessageVaultRestoreTask.Factory class that handles the injection for you?

Reply:
I create a dedicated service to restoring, and creating restoring task doesn't require to have knowledge about what components need for the restoring.